stdlib/public/core/EmptyCollection.swift

```diff
@@ -17,32 +17,42 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// An iterator that never produces an element.
+/// A collection whose element type is `Element` but that is always empty.
 @_fixed_layout // FIXME(sil-serialize-all)
-public struct EmptyIterator<Element> {
+public struct EmptyCollection<Element> {
   // no properties
-  
+
   /// Creates an instance.
   @_inlineable // FIXME(sil-serialize-all)
   public init() {}
 }
 
-extension EmptyIterator: IteratorProtocol, Sequence {
+extension EmptyCollection {
+  /// An iterator that never produces an element.
+  @_fixed_layout // FIXME(sil-serialize-all)
+  public struct Iterator {
```

Comment:
Could `EmptyCollection` just be its own `Iterator`?

Reply:
I guess. I think I prefer it consistent tho.